builds.go

```diff
@@ -0,0 +1,431 @@
+		return nil
+	}
+
+	for i := BuildPending; i <= BuildCanceled; i = i << 1 {
+		if (scopes & int(i)) != 0 {
+			temp := append(*opts.Scope, buildScope(i))
+			opts.Scope = &temp
+		}
+	}
+
+	if len(*opts.Scope) == 0 {
+		return nil
+	}
+
+	return opts
+}
+
+// BuildsService handles communication with the ci builds related methods
+// of the GitLab API.
+//
+// GitLab API docs: http://docs.gitlab.com/ce/api/builds.html
+type BuildsService struct {
+	client *Client
+}
+
+// Build represents a ci build.
+//
+// GitLab API docs: http://docs.gitlab.com/ce/api/builds.html
+type Build struct {
+	Commit *struct {
+		AuthorEmail string     `json:"author_email"`
+		AuthorName  string     `json:"author_name"`
+		CreatedAt   *time.Time `json:"created_at"`
+		ID          string     `json:"id"`
+		Message     string     `json:"message"`
+		ShortID     string     `json:"short_id"`
+		Title       string     `json:"title"`
+	} `json:"commit"`
+	Coverage      *string    `json:"coverage, omitempty"`
+	CreatedAt     *time.Time `json:"created_at"`
+	ArtifactsFile *struct {
+		Filename string `json:"filename"`
+		Size     int    `json:"size"`
+	} `json:"artifacts_file"`
+	FinishedAt *time.Time `json:"finished_at"`
+	ID         int        `json:"id"`
+	Name       string     `json:"name"`
+	Ref        string     `json:"ref"`
+	Runner     *struct {
+		ID          int    `json:"id"`
+		Description string `json:"description"`
+		Active      bool   `json:"active"`
+		IsShared    bool   `json:"is_shared"`
+		Name        string `json:"Name"`
+	} `json:"runner, omitempty"`
+	Stage     string     `json:"stage"`
+	StartedAt *time.Time `json:"started_at"`
+	Status    string     `json:"status"`
+	Tag       bool       `json:"tag"`
+	User      *struct {
+		AvatarURL  string     `json:"avatar_url"`
+		BIO        string     `json:"bio"`
+		CreatedAt  *time.Time `json:"created_at"`
+		ID         int        `json:"id"`
+		IsAdmin    bool       `json:"is_admin"`
+		Linkedin   string     `json:"linkedin"`
+		Name       string     `json:"name"`
+		Skype      string     `json:"skype"`
+		State      string     `json:"state"`
+		Twitter    string     `json:"twitter"`
+		Username   string     `json:"username"`
+		WebURL     string     `json:"web_url"`
+		WebsiteURL string     `json:"website_url"`
+	} `json:"user, omitempty"`
+}
+
+// ListProjectBuilds gets a list of builds in a project.
+//
+// The scope of builds to show, one or array of: pending, running,
+// failed, success, canceled; showing all builds if none provided.
+//
+// GitLab API docs:
+// http://docs.gitlab.com/ce/api/builds.html#list-project-builds
+func (s *BuildsService) ListProjectBuilds(pid interface{}, opts *ListBuildsOptions) ([]Build, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/builds", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	builds := []Build{}
```

Comment:
`var builds []Build` would be nicer here IMHO (goes for the other locations in this file as well of course)